unity-client/Assets/Rendering/Culling/CullingObjectsTracker.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Collections;
+using System.Linq;
+using UnityEngine;
+
+namespace DCL.Rendering
+{
+    public interface ICullingObjectsTracker
+    {
+        void SetDirty();
+        bool IsDirty();
+        Renderer[] GetRenderers();
+        SkinnedMeshRenderer[] GetSkinnedRenderers();
+        Animation[] GetAnimations();
+        IEnumerator PopulateRenderersList();
+    }
+
+    /// <summary>
+    /// This class is used for tracking all the renderers, skinnedMeshRenderers and Animations of the world.
+    ///
+    /// It currently uses a very lazy FindObjectsOfType approach, but is enough for its purposes as its used
+    /// to optimize a bigger bottleneck.
+    /// </summary>
+    public class CullingObjectsTracker : ICullingObjectsTracker
+    {
+        Renderer[] renderers;
+        SkinnedMeshRenderer[] skinnedRenderers;
+        Animation[] animations;
+
+        bool dirty = true;
+
+        /// <summary>
+        /// If the dirty flag is true, this coroutine will re-populate all the tracked objects.
+        /// </summary>
+        public IEnumerator PopulateRenderersList()
+        {
+            if (!dirty)
+                yield break;
+
+            renderers = Object.FindObjectsOfType<Renderer>()
```

Comment:
In the future we can probably track every renderer we create at that point in time and we would avoid calling FindObjectsOfType repeatedly.

Reply:
Absolutely. For the future, I think we may encapsulate all of this "renderer tracking" behaviour in a some kind of data store that fits the `MeshesInfo` specs, and have the `SceneBoundariesChecker` query that instead of `MeshesInfo`.